core/http-auth-spi/src/main/java/software/amazon/awssdk/http/auth/spi/signer/SignerProperty.java

```diff
@@ -30,36 +30,28 @@
 @Immutable
 @ThreadSafe
 public final class SignerProperty<T> {
-    private final Class<T> clazz;
+    private final String namespace;
     private final String name;
 
-    private SignerProperty(Class<T> clazz, String name) {
-        Validate.paramNotNull(clazz, "clazz");
+    private SignerProperty(String namespace, String name) {
+        Validate.paramNotBlank(namespace, "namespace");
         Validate.paramNotBlank(name, "name");
 
-        this.clazz = clazz;
+        this.namespace = namespace;
         this.name = name;
     }
 
     /**
-     * Create an instance of a property with a given type and name.
+     * Create a property of a certain type by specificying which class the property is being created in and name of property.
```

Comment:
There is a small chance that someone thinks the `Class` input parameter needs to be the "type" of the property. i.e., they call SignerProperty<Boolean> PROP1 = SignerProperty.create(Boolean.class, "prop1"). Might be better to javadoc `@param` for the 2 variables as well as <T> to guide the user.

I think the javadocs for namespace will get clearer when we add uniqueness and reflect that in the javadocs.